packages/wmr/src/lib/normalize-options.js

```diff
@@ -97,14 +96,81 @@ export async function normalizeOptions(options, mode) {
 
 	Object.defineProperty(options, '_config', { value: custom });
 	if (custom) {
-		if (custom.default) await custom.default(options);
-		if (custom[mode]) await custom[mode](options);
+		const fn = custom.default || custom[mode];
+		if (fn) {
+			const res = await fn(options);
+			if (res) {
+				options.plugins.push(res);
+			}
+		}
+	}
+
+	// Sort plugins by "enforce" phase. Default is "normal".
+	// The execution order is: "pre" -> "normal" -> "post"
+	if (options.plugins) {
+		options.plugins = options.plugins.flat().sort((a, b) => {
+			if (a.enforce === b.enforce) return 0;
+			else if ((a.enforce === 'pre' && b.enforce !== 'pre') || b.enforce === 'post') {
+				return -1;
+			} else if (b.enforce === 'pre' || a.enforce === 'post') {
+				return 1;
+			}
+			return 0;
+		});
 	}
 
+	/**
+	 * @param {keyof WMRPlugin} name
+	 */
+	const runConfigHook = name => {
+		options.plugins.forEach(plugin => {
+			if (!plugin[name]) return;
+
+			const res = plugin[name](options);
```

Comment:
Do you know if Vite's version of this allows async? I'm fine with this being synchronous, just not sure if they bothered.

Reply:
Good call. They allow everything to be async which makes sense. Will update.